ui/dashboards/src/stories/decorators/WithDashboard.tsx

```diff
@@ -15,6 +15,12 @@ import { StoryFn, StoryContext } from '@storybook/react';
 import { DashboardProvider, DashboardProviderProps } from '@perses-dev/dashboards';
 import { EMPTY_DASHBOARD_RESOURCE } from './constants';
 
+declare module '@storybook/react' {
```

Comment:
This is the best way I could find to type the parameters for now based on looking at a few discussions/issues on the storybook github. It seems to work well as long as we don't have duplicates, which is why I renamed the small number of decorators with duplicate names.